rslib/src/sync/http_server/mod.rs

```diff
@@ -91,9 +96,26 @@ impl SimpleServerInner {
             match std::env::var(&envvar) {
                 Ok(val) => {
                     let hkey = derive_hkey(&val);
-                    let (name, _) = val.split_once(':').with_whatever_context(|| {
-                        format!("{envvar} should be in 'username:password' format.")
-                    })?;
+                    let (name, pwhash) = {
+                        let (name, password) = val.split_once(':').with_whatever_context(|| {
+                            format!("{envvar} should be in 'username:password' format.")
+                        })?;
+                        match std::env::var("PASSWORDS_HASHED") {
+                            Ok(_) => (name, password.to_string()),
```

Comment:
This match statement can be more simply expressed with` if env::var("PASSWORDS_HASHED").is_ok() {`